lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -912,29 +912,46 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     }
     return hasPrintedAnything;
   }
-    
-  void printRenameForDecl(const AvailableAttr *AvAttr, const ValueDecl *D,
-                          bool includeQuotes) {
-    assert(!AvAttr->Rename.empty());
-    
-    auto renamedParsedDeclName = parseDeclName(AvAttr->Rename);
-    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
-    
+  
+  const ValueDecl *getRenameDecl(const ValueDecl *D, const ParsedDeclName renamedParsedDeclName) {
     auto declContext = D->getDeclContext();
-    const ValueDecl *renamedDecl = nullptr;
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
     
     if (isa<ClassDecl>(D) || isa<ProtocolDecl>(D)) {
       UnqualifiedLookup lookup(renamedDeclName.getBaseIdentifier(),
                                declContext->getModuleScopeContext(),
                                nullptr,
                                SourceLoc(),
                                UnqualifiedLookup::Flags::TypeLookup);
-      renamedDecl = lookup.getSingleTypeResult();
+      return lookup.getSingleTypeResult();
     } else {
+      TypeDecl *typeDecl = declContext->getSelfNominalTypeDecl();
+      
+      if (!renamedParsedDeclName.ContextName.empty()) {
+        auto contextIdentifier = D->getASTContext().getIdentifier(renamedParsedDeclName.ContextName);
+        UnqualifiedLookup specificTypeLookup(contextIdentifier,
+                                             declContext->getModuleScopeContext(),
+                                             nullptr,
+                                             SourceLoc(),
+                                             UnqualifiedLookup::Flags::TypeLookup);
+        if (!specificTypeLookup.getSingleTypeResult()) {
+          return nullptr;
+        }
+        
+        if (typeDecl->getDeclaredInterfaceType()->matches(specificTypeLookup.getSingleTypeResult()->getDeclaredInterfaceType(), TypeMatchFlags::AllowOverride)) {
```

Comment:
Nitpick: 80-column limit (here and below). It's okay to break this into multiple expressions.